rmq/consume.go

```diff
@@ -78,7 +80,25 @@ func NewConsumer(amqpURI, exchange, queue, routingKey, tag string, prefetch int)
 
 func (r *RabbitMQ) ackMultiple(deliveryTag <-chan Verify) {
 	for v := range deliveryTag {
-		r.channel.Ack(v.Tag, v.MultiAck)
+		err := r.channel.Ack(v.Tag, v.MultiAck)
+		if err != nil {
+			log.Fatalf("rabbit channel ack failed %s", err)
+		}
+	}
+}
+
+func (r *RabbitMQ) SafeStop() {
+	err := r.channel.Cancel(
+		r.tag, // name
+		false, // noWait
+	)
+	if err != nil {
+		log.Fatalf("rabbit channel cancel failed %s", err)
+	} else {
+		for _ = range time.Tick(2 * time.Second) {
```

Comment:
why do we need the time tick here?

Reply:
It's a "crutch", to allow for all AMQP-messages to be put into the write channel before it closes. I'm not so familiar with the golang, to avoid time tick/sleep usage, but it looks like a better approach exists.